<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.9.2" />
<title>protein API documentation</title>
<meta name="description" content="This is Part4: Get the information of protein sequence of the gene" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>protein</code></h1>
</header>
<section id="section-intro">
<p>This is Part4: Get the information of protein sequence of the gene</p>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">&#34;&#34;&#34;This is Part4: Get the information of protein sequence of the gene &#34;&#34;&#34;
import matplotlib.pylab as plt
import numpy as np

def protein_sequence(seq):
    &#34;&#34;&#34;Return the information of protein sequence of target gene.&#34;&#34;&#34;
    #get the seq from part1 Download function
    #Providing the amino acid background information by using dictory
    aa_dic = {&#34;A&#34; : [&#34;GCT&#34;,&#34;GCC&#34;,&#34;GCA&#34;,&#34;GCG&#34;],
        &#34;L&#34; : [&#34;TTA&#34;,&#34;TTG&#34;,&#34;CTT&#34;,&#34;CTC&#34;,&#34;CTA&#34;,&#34;CTG&#34;],
        &#34;R&#34; : [&#34;CGT&#34;,&#34;CGC&#34;,&#34;CGA&#34;,&#34;CGG&#34;,&#34;AGA&#34;,&#34;AGG&#34;],
        &#34;K&#34; : [&#34;AAA&#34;,&#34;AAG&#34;],&#34;N&#34; : [&#34;AAT&#34;,&#34;AAC&#34;],
        &#34;M&#34; : [&#34;ATG&#34;],&#34;D&#34; : [&#34;GAT&#34;,&#34;GAC&#34;],
        &#34;F&#34; : [&#34;TTT&#34;,&#34;TTC&#34;],&#34;C&#34; : [&#34;TGT&#34;,&#34;TGC&#34;],
        &#34;P&#34; : [&#34;CCT&#34;,&#34;CCC&#34;,&#34;CCA&#34;,&#34;CCG&#34;],
        &#34;Q&#34; : [&#34;CAA&#34;,&#34;CAG&#34;],
        &#34;S&#34; : [&#34;TCT&#34;,&#34;TCC&#34;,&#34;TCA&#34;,&#34;TCG&#34;,&#34;AGT&#34;,&#34;AGC&#34;],
        &#34;E&#34; : [&#34;GAA&#34;,&#34;GAG&#34;],
        &#34;T&#34; : [&#34;ACT&#34;,&#34;ACC&#34;,&#34;ACA&#34;,&#34;ACG&#34;],
        &#34;G&#34; : [&#34;GGT&#34;,&#34;GGC&#34;,&#34;GGA&#34;,&#34;GGG&#34;],
        &#34;W&#34; : [&#34;TGG&#34;],&#34;H&#34; : [&#34;CAT&#34;,&#34;CAC&#34;],
        &#34;Y&#34; : [&#34;TAT&#34;,&#34;TAC&#34;],
        &#34;I&#34; : [&#34;ATT&#34;,&#34;ATC&#34;,&#34;ATA&#34;],
        &#34;V&#34; : [&#34;GTT&#34;,&#34;GTC&#34;,&#34;GTA&#34;,&#34;GTG&#34;]}
    name = [&#34;A&#34;,&#34;L&#34;,&#34;R&#34;,&#34;K&#34;,&#34;N&#34;,&#34;M&#34;,&#34;D&#34;,&#34;F&#34;,&#34;C&#34;,&#34;P&#34;,&#34;Q&#34;,&#34;S&#34;,&#34;E&#34;,&#34;T&#34;,&#34;G&#34;,&#34;W&#34;,&#34;H&#34;,&#34;Y&#34;,&#34;I&#34;,&#34;V&#34;]
    #getting the protein sequence
    res_p = []
    start = seq.index(&#34;ATG&#34;)+3
    for i in range(start,len(seq),3):
        sed = seq[i:i+3]
        for j in name:
            if sed in aa_dic[j]:
                posit = name.index(j)
        res_p.append([key for key in aa_dic.keys()][posit])
        proteinseq = &#39;&#39;.join(res_p)
    return proteinseq
    #count the amino acid number
    count = []
    for i in name:
        count.append(proteinseq.count(i))
    ind = list(range(1, 21))
    n_num = count
    _, ax_x = plt.subplots()
    ax_x.scatter(ind, count)
    for i, txt in enumerate(n_num):
        ax_x.annotate(txt, (ind[i], count[i]))
    plt.xticks(ind,name)
    plt.ylabel(&#39;Count&#39;)
    plt.title(&#39;Amino Acid Distribution&#39;)
    plt.xlim()
    plt.ylim(-1,np.max(count)+5)
    plt.plot(ind, count,color=&#34;green&#34;)</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="protein.protein_sequence"><code class="name flex">
<span>def <span class="ident">protein_sequence</span></span>(<span>seq)</span>
</code></dt>
<dd>
<div class="desc"><p>Return the information of protein sequence of target gene.</p></div>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def protein_sequence(seq):
    &#34;&#34;&#34;Return the information of protein sequence of target gene.&#34;&#34;&#34;
    #get the seq from part1 Download function
    #Providing the amino acid background information by using dictory
    aa_dic = {&#34;A&#34; : [&#34;GCT&#34;,&#34;GCC&#34;,&#34;GCA&#34;,&#34;GCG&#34;],
        &#34;L&#34; : [&#34;TTA&#34;,&#34;TTG&#34;,&#34;CTT&#34;,&#34;CTC&#34;,&#34;CTA&#34;,&#34;CTG&#34;],
        &#34;R&#34; : [&#34;CGT&#34;,&#34;CGC&#34;,&#34;CGA&#34;,&#34;CGG&#34;,&#34;AGA&#34;,&#34;AGG&#34;],
        &#34;K&#34; : [&#34;AAA&#34;,&#34;AAG&#34;],&#34;N&#34; : [&#34;AAT&#34;,&#34;AAC&#34;],
        &#34;M&#34; : [&#34;ATG&#34;],&#34;D&#34; : [&#34;GAT&#34;,&#34;GAC&#34;],
        &#34;F&#34; : [&#34;TTT&#34;,&#34;TTC&#34;],&#34;C&#34; : [&#34;TGT&#34;,&#34;TGC&#34;],
        &#34;P&#34; : [&#34;CCT&#34;,&#34;CCC&#34;,&#34;CCA&#34;,&#34;CCG&#34;],
        &#34;Q&#34; : [&#34;CAA&#34;,&#34;CAG&#34;],
        &#34;S&#34; : [&#34;TCT&#34;,&#34;TCC&#34;,&#34;TCA&#34;,&#34;TCG&#34;,&#34;AGT&#34;,&#34;AGC&#34;],
        &#34;E&#34; : [&#34;GAA&#34;,&#34;GAG&#34;],
        &#34;T&#34; : [&#34;ACT&#34;,&#34;ACC&#34;,&#34;ACA&#34;,&#34;ACG&#34;],
        &#34;G&#34; : [&#34;GGT&#34;,&#34;GGC&#34;,&#34;GGA&#34;,&#34;GGG&#34;],
        &#34;W&#34; : [&#34;TGG&#34;],&#34;H&#34; : [&#34;CAT&#34;,&#34;CAC&#34;],
        &#34;Y&#34; : [&#34;TAT&#34;,&#34;TAC&#34;],
        &#34;I&#34; : [&#34;ATT&#34;,&#34;ATC&#34;,&#34;ATA&#34;],
        &#34;V&#34; : [&#34;GTT&#34;,&#34;GTC&#34;,&#34;GTA&#34;,&#34;GTG&#34;]}
    name = [&#34;A&#34;,&#34;L&#34;,&#34;R&#34;,&#34;K&#34;,&#34;N&#34;,&#34;M&#34;,&#34;D&#34;,&#34;F&#34;,&#34;C&#34;,&#34;P&#34;,&#34;Q&#34;,&#34;S&#34;,&#34;E&#34;,&#34;T&#34;,&#34;G&#34;,&#34;W&#34;,&#34;H&#34;,&#34;Y&#34;,&#34;I&#34;,&#34;V&#34;]
    #getting the protein sequence
    res_p = []
    start = seq.index(&#34;ATG&#34;)+3
    for i in range(start,len(seq),3):
        sed = seq[i:i+3]
        for j in name:
            if sed in aa_dic[j]:
                posit = name.index(j)
        res_p.append([key for key in aa_dic.keys()][posit])
        proteinseq = &#39;&#39;.join(res_p)
    return proteinseq
    #count the amino acid number
    count = []
    for i in name:
        count.append(proteinseq.count(i))
    ind = list(range(1, 21))
    n_num = count
    _, ax_x = plt.subplots()
    ax_x.scatter(ind, count)
    for i, txt in enumerate(n_num):
        ax_x.annotate(txt, (ind[i], count[i]))
    plt.xticks(ind,name)
    plt.ylabel(&#39;Count&#39;)
    plt.title(&#39;Amino Acid Distribution&#39;)
    plt.xlim()
    plt.ylim(-1,np.max(count)+5)
    plt.plot(ind, count,color=&#34;green&#34;)</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="protein.protein_sequence" href="#protein.protein_sequence">protein_sequence</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc"><cite>pdoc</cite> 0.9.2</a>.</p>
</footer>
</body>
</html>